cmd/topicmappr/commands/reassignments.go

```diff
@@ -288,6 +288,55 @@ func computeReassignmentBundles(
 	return results
 }
 
+/**
+getPartitionMapChunk Breaks a reassignment into a series of sequential, smaller reassignments.
+For large reassignments that may take a while, or risky operations that may require downtime in between, a chunked reassignment can be used.
+This will generate a series of partition maps that will converge on the desired state. To minimize data transfer,
+partitions are only moved to replicas in the desired final state.
+
+The original design was intended for downscaling operations, to remove partitions from one (or three) brokers at a time,
+without overwhelming whatever brokers are remaining in the cluster.
+*/
+func getPartitionMapChunk(finalMap *kafkazk.PartitionMap, initialMap *kafkazk.PartitionMap, brokerIds kafkazk.BrokerList, chunkStepSize int) []*kafkazk.PartitionMap {
+	var intermediateMap = initialMap.Copy()
+	var out []*kafkazk.PartitionMap
+	brokerIds.SortByIDDesc()
+
+	for i := 0; i < len(brokerIds); i += chunkStepSize {
+		// Select the brokers we will move data from for this chunk
+		var chunkBrokers = map[int]struct{}{}
+		for j := 0; j < chunkStepSize && i+j < len(brokerIds); j++ {
```

Comment:
Just eyeing this, will we skip chunking if we have say 3 brokers and specify a chunk size of 5?

Reply:
I'll add a test case for that

Reply:
(actually this may or may not belong to L305)